src/machine.cpp

```diff
@@ -527,7 +528,10 @@ machine::machine(const machine_config &c, const machine_runtime_config &r) :
     }
 
     // Initialize TTY if console input is enabled
-    if (m_c.htif.console_getchar) {
+    if (m_c.htif.console_getchar || has_virtio_console()) {
```

Comment:
Can you still set both htif.console_getchar and add a virtio console? Shouldn't we forbid this?

Reply:
Yes you can set both. I don't think we should forbid, so if the OS fails to initialize VirtIO it can fallback to HTIF console if enabled. I don't see issues having both enabled, this is something I did for a while.